### ISSUE TITLE
[DGSM] Bug: Editting multiple duplicant properties cause crash

### ISSUE THREAD

**Issue by StrikerCodes** [O-n-y/oxy.mods#137]
# Description

After worldgen, at initial duplicant selection, editing multiple duplicants causes the game to crash.

# Expected behavior

Opening the edit properties screen of multiple duplicants and editing them makes changes to the duplicants, and pressing the apply changes button applies the change to the duplicants to be spawned.

# Actual behavior

Opening the edit properties screen on multiple duplicants causes the apply changes button becomes unresponsive on the older duplicant, and pressing any other buttons in the screen leads to game crash.

# Steps to reproduce

1. Start a normal game
2. continue until starting duplicants screen
3. click edit duplicant on 1st and 2nd duplicant
4. click apply changes to 2nd duplicant
5. apply changes on first duplicant becomes unresponsive (bug)
6. clicking any other buttons leads to game crash (bug)

![step_1](https://github.com/O-n-y/oxy.mods/assets/132714754/a32a340a-f79c-47be-bf7b-e3fd3f5960bf)
![step_2](https://github.com/O-n-y/oxy.mods/assets/132714754/7582e1da-b8c3-4509-abf8-6ba6de19fb9f)
![step_3](https://github.com/O-n-y/oxy.mods/assets/132714754/fc7a8cac-95f4-459a-98cb-2bb83012f4c2)
![step_4](https://github.com/O-n-y/oxy.mods/assets/132714754/5538e60b-3326-4e5a-83ce-bc04c933d19e)
![step_5](https://github.com/O-n-y/oxy.mods/assets/132714754/3ddc71e2-334b-4e21-84fd-7d4782503ec7)

**Comment by DMFan79** [comment]
I'm having the exact same problem. This is what the game shows after the crash:

```
NullReferenceException

Ony.OxygenNotIncluded.Tools.ShowDialogMenu (System.String name, System.String title, UnityEngine.GameObject owner, System.Action`2[T1,T2] addButtons, System.Int32 height, Ony.Unity.UI.ScrollPosition scrollPosition) (at <f7900d98c6fe46e5b62b12c9e2b3c9bb>:0)
Ony.OxygenNotIncluded.DGSM2.CharacterHelper+<>c__DisplayClass10_2.<GenerateTraitsView>b__0 (KKeyCode code) (at <f7900d98c6fe46e5b62b12c9e2b3c9bb>:0)
KButton.SignalClick (KKeyCode btn) (at <1e2b7d5db95c4d6b84eabb7ca0270927>:0)
KButton.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at <1e2b7d5db95c4d6b84eabb7ca0270927>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at <e9635660f69b4d9d84c085aeda3be353>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at <e9635660f69b4d9d84c085aeda3be353>:0)
UnityEngine.EventSystems.EventSystem:Update()

Build: U50-583750-S
```

**Comment by taozhiyu** [comment]
crash Video here:

(ignore other mods, I‘m sure not caused by them)

https://github.com/O-n-y/oxy.mods/assets/49258735/511406b1-a9cd-4921-b416-ba276438b768

**Comment by Murkyflurky** [comment]
Can confirm this happens to me with specifically only Ony mods (No drop and the mod manager besides DGSM), and mod version showed as: 3.6.2.16
And game build shows as: 587362-5

**Comment by O-n-y** [comment]
Will be fixed on upcoming release version  3.6.7.37+ (currently in BETA via patrons)

**Comment by O-n-y** [comment]
fixed
